onebusaway-android/src/main/java/org/onebusaway/android/ui/NavigationDrawerFragment.java

```diff
@@ -375,23 +508,91 @@ private void createNavDrawerItems() {
 
         LinearLayout containerLayout = (LinearLayout) mDrawerItemsListContainer.
                 findViewById(R.id.navdrawer_items_list);
-
         containerLayout.removeAllViews();
 
-        for (int itemId : mNavDrawerItems) {
-            mNavDrawerItemViews[i] = makeNavDrawerItem(itemId, containerLayout);
-            containerLayout.addView(mNavDrawerItemViews[i]);
-            ++i;
+        ObaRegion currentRegion = Application.get().getCurrentRegion();
+        if (currentRegion != null && currentRegion.getSupportsEmbeddedSocial() && isSignedIn) {
+            // user is signed in to Embedded Social
+            createSocialOverflow(containerLayout);
+            for (int itemId : mNavDrawerItems) {
+                mNavDrawerItemViews[i] = makeNavDrawerItem(itemId, containerLayout);
+                if (isSocialOverflow(itemId)) {
+                    // add to the overflow section
+                    socialOverflowContainer.addView(mNavDrawerItemViews[i]);
+                } else {
+                    containerLayout.addView(mNavDrawerItemViews[i]);
+                }
+                ++i;
+            }
+
+            // add the overflow menu
+            containerLayout.addView(socialOverflowButtonWrapper, OVERFLOW_INDEX);
+            containerLayout.addView(socialOverflowContainer, OVERFLOW_INDEX + 1);
+        } else {
+            // user is not signed in
+            for (int itemId : mNavDrawerItems) {
+                mNavDrawerItemViews[i] = makeNavDrawerItem(itemId, containerLayout);
+                containerLayout.addView(mNavDrawerItemViews[i]);
+                ++i;
+            }
+        }
+    }
+
+    private void createSocialOverflow(ViewGroup parent) {
+        SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(getActivity());
```

Comment:
Please use `Application.get().getPrefs()` everywhere in the app to get the shared preferences.  Strange things can happen depending on exactly how you get an instance of this, and we run everything through the same method to ensure consistency.